### PR TITLE
Lesson Cards Are Scrollable

### PR DESCRIPTION
**Pull request by sandroroux**
Closes [asteroid/#44](https://github.com/catalpainternational/asteroid/issues/44)

### CODE REVIEW

File: src/riot/Components/TopMenu.riot.html [hunk 1]
```diff
@@ -66,25 +50,28 @@
...
+                    this.state.isProfile || this.state.isNotifications || this.state.isSettings;
 
                 if (this.state.isProfile) {
                     this.state.color = "profile-green";
                 } else if (this.state.isNotifications) {
                     this.state.color = "notifications-orange";
-                } else if (this.state.isHome) {
+                } else if (this.state.isHome || this.state.isCard) {
                     this.state.color = "blue-1";
                 } else {
```
Comment (sandroroux) [inline src/riot/Components/TopMenu.riot.html:71]: @katevoss Sorry, my auto-formatter exploded the change list. These lines are new.

File: src/scss/_lesson.scss [hunk 2]
```diff
@@ -238,21 +245,24 @@ LessonFrame {
...
+        height: $screen-height-minus-top-menu;
         max-width: 40em;
 
         background-color: $white;
+
+        @media only screen and (max-width: $sm-phone) {
+            height: auto;
+        }
     }
 }
```
Comment (sandroroux) [inline src/scss/_lesson.scss:262]: This style rule (`LessonFrame {`) is the other majorly changed area.

File: src/scss/_lesson.scss [hunk 3]
```diff
@@ -210,23 +210,30 @@ LessonPage .description {
...
 
         .card-frame {
-            min-height: auto;
+            // min-height: auto;
         }
     }
 }
 
 LessonFrame {
+    $screen-height-minus-top-menu: calc(100% - #{$top-menu-height});
```
Comment (katevoss) [inline src/scss/_lesson.scss:228]: Can you describe what the value of `$screen-height-minus-top-menu` represents, and how does it differ from `$bottom-menu-height-no-footer`?

File: src/scss/_lesson.scss [hunk 4]
```diff
@@ -210,23 +210,30 @@ LessonPage .description {
...
+
     .frame-background {
-        height: calc(100vh - #{$top-menu-height-no-footer});
-        margin-top: -#{$top-menu-height};
-        padding-top: calc(#{$top-menu-height} - 10px);
+        height: $screen-height-minus-top-menu;
+
+        @media only screen and (max-width: $sm-phone) {
+            height: auto;
```
Comment (katevoss) [inline src/scss/_lesson.scss:234]: Why does `height: auto` not work when the screen is wider than `$sm-phone`? I'm guessing it's a bootstrap thing.

Comment (sandroroux) [reply src/scss/_lesson.scss:234]: I think `height: auto` ensures the card fits the contents. When the screen is taller than the card, the `height: auto` card doesn't fill the screen.

On small phones, the `height: auto` card is taller than the screen. The screen becomes scrollable.

In explaining that, I think `height: $screen-height-minus-top-menu;` should be `min-height: $screen-height-minus-top-menu;`

File: src/scss/_lesson.scss (hunk 3 above)
Comment (sandroroux) [reply src/scss/_lesson.scss:228]: Originally, Sergio used negative margins to add color to the top menu. When all elements of the page render with defaults, the top menu appears with the card-background underneath. The negative margin pulled the background under the transparent top menu.

`$screen-height-minus-top-menu` calculates the height between the top menu's bottom and the screen's bottom. The card should sit in that space.

I don't know what the name `$bottom-menu-height-no-footer` suggests. I thought the bottom menu was the old overlay (home button, notifications button, profile button). The bottom menu should always have a footer. My confusion arises from that.

Long story ended, I created my own variable because I'm confused.

Comment (katevoss) [reply src/scss/_lesson.scss:228]: I must admit that I added to your confusion as I added the `$bottom-menu-height-no-footer` variable, and I regrettably confused myself in my poor naming! In https://github.com/catalpainternational/canoe/blob/RC-1.0/src/scss/_variables.scss#L44 I recall that I added the variable `$bottom-menu-height-no-footer` when I removed the bottom menu (home, notifications, profile) from the cards (objective, content, test). Maybe the variable should be called `$bottom-footer-height-no-menu` to more accurately describe it.

**Comment by sandroroux** [comment]
image

image

image

**Comment by PeteCoward** [comment]
@katevoss @sandroroux please resolve the conflicts with this and decide if we should merge. I'm kinda lost with all the back and forth.

**Comment by sandroroux** [comment]
@PeteCoward, sounds good. I'll merge this on top of @katevoss's multiple-answer PR.

**Comment by PeteCoward** [comment]
@katevoss @sandroroux please resolve the conflicts with this and decide if we should merge. I'm still kinda lost with all the back and forth.

**Comment by sandroroux** [comment]
@PeteCoward, @katevoss  I resolved the conflicts, but the PR is a little janky. I'll smooth it tomorrow.

File: src/riot/Components/LessonFrame.riot.html [hunk 5]
```diff
@@ -1,7 +1,9 @@
 <LessonFrame>
     <div class="frame-background background-blue-1">
         <div class="card-frame">
-            <slot />
+            <div class="card-container {props.extrastyleclasses}">
```
Comment (katevoss) [inline src/riot/Components/LessonFrame.riot.html:4]: @sandroroux why did you add the `card-container` to the `LessonFrame`?
It causes an issue in LessonTest, the answers are centered with `align-items: center`
image

File: src/riot/Lesson/LessonTest.riot.html [hunk 6]
```diff
@@ -28,6 +26,7 @@ <h3>{state.cardIdx + 1}. {state.cards[state.cardIdx].question}</h3>
         module="{props.module}"
         cardIdx="{state.cardIdx}"
     ></TestPopupModal>
+    </LessonFrame>
```
Comment (katevoss) [inline src/riot/Lesson/LessonTest.riot.html:29]: @sandroroux Did you mean to add two `LessonFrame` closing tags? Here on line 29 and on line 17. I only see one opening tag on line 2.

File: src/riot/Lesson/LessonTest.riot.html [hunk 7]
```diff
@@ -90,12 +91,13 @@ <h3>{state.cardIdx + 1}. {state.cards[state.cardIdx].question}</h3>
...
                 const card = this.state.cards[cardIdx];
-                const correctAnswers = card && card.answers.filter(answer => answer.correct);
+                const correctAnswers = card && card.answers.filter((answer) => answer.correct);
                 const isSingleAnswer = card && correctAnswers.length === 1;
 
                 this.update({
                     cardIdx,
-                    isSingleAnswer
+                    isSingleAnswer,
+                    cardIdx: getLessonCardIdx(),
```
Comment (katevoss) [inline src/riot/Lesson/LessonTest.riot.html:100]: @sandroroux There are two `cardIdx` updates here, online 98 and 100. Is that intentional?

Comment (sandroroux) [reply src/riot/Lesson/LessonTest.riot.html:102]: Good catch. That's a mistake.

**Comment by sandroroux** [comment]
@katevoss, @PeteCoward, I fixed the conflicts (by rebasing — delete your branch and `checkout` the latest version). Do you mind reviewing this?
